forch/radius_query.py

```diff
@@ -0,0 +1,113 @@
+                 server_port):
+        self.socket = None
+        self.listen_ip = listen_ip
+        self.listen_port = listen_port
+        self.server_ip = server_ip
+        self.server_port = server_port
+
+    def setup(self):
+        """Setup RADIUS Socket"""
+        LOGGER.info("Setting up radius socket.")
+        try:
+            self.socket = socket.socket(socket.AF_INET,
+                                        socket.SOCK_DGRAM)
+            self.socket.bind((self.listen_ip, self.listen_port))
+        except socket.error as err:
+            self.logger.error("Unable to setup socket: %s", str(err))
+            raise err
+
+    def send(self, data):
+        """Sends on the radius socket
+            data (bytes): what to send"""
+        self.socket.sendto(data, (self.server_ip, self.server_port))
+
+    def receive(self):
+        """Receives from the radius socket"""
+        return self.socket.recv(4096)
+
+class RadiusQuery:
+    """Maintains socket information and sends out and receives requests form RADIUS server"""
+    def __init__(self, socket_info, radius_secret):
+        self.next_radius_id = 0
+        self.packet_id_to_mac = {}
+        self.packet_id_to_req_authenticator = {}
+        self.running = True
+        self.radius_output_q = Queue()
+        #TODO: Find better way to handle secret
+        self.radius_secret = radius_secret
+        self.radius_socket = RadiusSocket(socket_info.listen_ip, socket_info.listen_port, \
+                                          socket_info.server_ip, socket_info.server_port)
+        self.radius_socket.setup()
+        #self.receive_radius_messages()
+
+    def receive_radius_messages(self):
+        while self.running:
+            LOGGER.info("Waiting for RADIUS messages.")
+            packed_message = self.radius_socket.receive()
+            try:
+                radius = self.decode_radius_response(packed_message)
+            except MessageParseError as exception:
+                LOGGER.warning("exception: %s. message: %s", packed_message, exception)
+            LOGGER.info("Received RADIUS msg: Code:%s packet_id:%s attributes:%s", radius.CODE, radius.packet_id, radius.attributes.to_dict())
+
+    def send_mab_request(self, src_mac, port_id):
+        req_packet = self.encode_mab_message(src_mac, port_id)
+        LOGGER.info("Sending MAB request for mac %s", src_mac)
+        self.radius_socket.send(req_packet)
+
+    def encode_mab_message(self, src_mac, port_id=None):
+        radius_id = self.get_next_radius_pkt_id()
+        req_authenticator = self.get_req_authenticator()
+        self.packet_id_to_mac[radius_id] = {'src_mac': src_mac, 'port_id': port_id}
+        self.packet_id_to_req_authenticator[radius_id] = req_authenticator
+
+        attr_list = []
+        mac_str = str(src_mac).replace(':', "")
+        attr_list.append(UserName.create(mac_str))
+        attr_list.append(CallingStationId.create(str(src_mac).replace(':', '-')))
```

Comment:
extract to variable like "mac_str" -- and neither one should be called mac_str, qualify each one (mac_username & mac_password?), or inline both.

Reply:
It is used thrice in the method. Are you sure I should name it different things? Because it is exactly what it is named, MAC addr in str format.

Reply:
I will inline it. It is twice not thrice.

Reply:
Done